### PR TITLE
Self articles actions

### PR DESCRIPTION
**Pull request by rRaijin**

### CODE REVIEW

File: api/article/views.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 from rest_framework import viewsets, permissions
+from rest_framework.decorators import detail_route
+
 
 from api.article.serializers import *
```
Comment (lerdem) [inline api/article/views.py]: That is a bad way to import smt with *. Better way is `from api.article import serializers`

File: api/article/views.py [hunk 2]
```diff
@@ -17,3 +34,15 @@ def perform_create(self, serializer, format=None):
...
+
+    def perform_update(self, serializer):
+        if self.request.data.get('image') is not None:
+            image = self.request.data.get('image')
+            serializer.save(image=image)
+        else:
+            serializer.save()
+
+    # TODO permissions author only
+    @detail_route(methods=['get'], permission_classes=[permissions.IsAuthenticated])
```
Comment (lerdem) [inline api/article/views.py:48]: Also you can manage you `permission_classes`  using them as `property`.
```python
@property
def  permission_classes(self):
    if self.action == 'self_articles':
        return articles.filter(author=author)
    return foo
```

File: api/article/views.py [hunk 3]
```diff
@@ -7,8 +9,23 @@ class ArticleViewSet(viewsets.ModelViewSet):
...
+        'partial_update': ArticleSerializer,
+    }
+
+    def get_serializer_class(self):
+        return self.serializer_class_map[self.action]
+
     def get_queryset(self):
-        return Article.objects.prefetch_related('author').order_by('-created')
+        author = self.request.user
+        articles = Article.objects.prefetch_related('author').order_by('-created')
```
Comment (lerdem) [inline api/article/views.py]: Perhaps better way is to move `order_by` as default ordering to `Article`. What do you think?

File: api/article/views.py [hunk 4]
```diff
@@ -7,8 +9,23 @@ class ArticleViewSet(viewsets.ModelViewSet):
     serializer_class = ArticleSerializer
     permission_classes = permissions.IsAuthenticatedOrReadOnly,
 
+    serializer_class_map = {
+        'retrieve': ArticleSerializer,
+        'list': ArticleSerializer,
+        'create': ArticleSerializer,
+        'self_articles': ArticleSerializer,
```
Comment (lerdem) [inline api/article/views.py]: Name it just `atricles`, i think it will be enough.

File: api/article/views.py [hunk 5]
```diff
@@ -7,8 +9,23 @@ class ArticleViewSet(viewsets.ModelViewSet):
     serializer_class = ArticleSerializer
```
Comment (lerdem) [inline api/article/views.py]: Do you still need that?

File: api/users/serializers.py [hunk 6]
```diff
@@ -18,18 +17,11 @@ def get_img_name(self, obj):
...
 
-    def get_short_description(self, obj):
-        if obj.description:
-            return obj.description[:100]
-        else:
-            return False
-
     class Meta:
         model = Article
         fields = (
```
Comment (lerdem) [inline api/users/serializers.py:22]:
```suggestion
        fields = read_only_fields (
```

File: api/article/views.py (hunk 5 above)
Comment (lerdem) [reply api/article/views.py]: Wait a minute. Why are you using `serializer_class_map` at all?

**Review by lerdem — COMMENTED** [review]
I did not look at the react code. The rest is fine for me.

File: api/article/views.py (hunk 2 above)
Comment (rRaijin) [reply api/article/views.py:48]: I plan add a custom permissions later

File: api/article/views.py (hunk 1 above)
Comment (rRaijin) [reply api/article/views.py]: Yes, I forgot to fix it here.

File: api/article/views.py (hunk 3 above)
Comment (rRaijin) [reply api/article/views.py]: Yes, that's right, it was a temporary solution.

File: api/article/views.py (hunk 4 above)
Comment (rRaijin) [reply api/article/views.py]: This endpoint is already taken. I'll think about changing the name in the future.

File: api/article/views.py (hunk 5 above)
Comment (rRaijin) [reply api/article/views.py]: OMG ))))))))))))))))))

**Pull request merged**